### PR TITLE
cross platform: replace long to int32

### PR DESCRIPTION
**Pull request by obastemur**
This PR replaces most of the `long` usage to `int32`. On some places `long`
type is kept as is or replaced to `LONG`.

### CODE REVIEW

**Comment by jianchun** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_debug please
@dotnet-bot test Ubuntu ubuntu_linux_release please

File: lib/Runtime/Base/ThreadContext.h [hunk 1]
```diff
@@ -753,7 +753,7 @@ class ThreadContext sealed :
     // Number of script context attached with probe manager.
     // This counter will be used as addref when the script context is created, this way we maintain the life of diagnostic object.
     // Once no script context available , diagnostic will go away.
-    long crefSContextForDiag;
+    LONG crefSContextForDiag;
```
Comment (jianchun) [inline lib/Runtime/Base/ThreadContext.h]: Why use `LONG` here? `ThreadContext.cpp` has `int32 lref = InterlockedDecrement(&crefSContextForDiag)`.

**Comment by jianchun** [comment]
LGTM other than one comment. Please fix merge conflict.

File: lib/Runtime/Base/ThreadContext.h (hunk 1 above)
Comment (obastemur) [inline lib/Runtime/Base/ThreadContext.h]: InterlockedDecrement complains for int32 + This is code is mostly Windows specific and we are kind of in peace with rest of the LONG/ULONG

Comment (jianchun) [inline lib/Runtime/Base/ThreadContext.h]: I see, then maybe `ThreadContext.cpp` should have 'LONG lref = InterlockedDecrement(...)`?

**Comment by obastemur** [comment]
@digitalinfinity I guess the plan is to merge this one after `merging linux to master or master to linux` right? Trying to figure out the best rebase moment.

**Comment by obastemur** [comment]
ping! Re-based this one to latest

**Comment by jianchun** [comment]
@obastemur LGTM other than that you accidentally added a bunch of `.orig` files.

**Comment by obastemur** [comment]
@jianchun thanks for the review.

**Pull request merged**